File: fornituras-api/src/main/java/com/numobiz/solutions/fornituras/common/dto/ApiResponse.java
package com.numobiz.solutions.fornituras.common.dto;

public record ApiResponse<T>(boolean success, String message, T data) {

	public static <T> ApiResponse<T> ok(T data) {
		return new ApiResponse<>(true, "operation successful", data);
	}

	public static <T> ApiResponse<T> ok(String message, T data) {
		return new ApiResponse<>(true, message, data);
	}

	public static ApiResponse<Void> ok(String message) {
		return new ApiResponse<>(true, message, null);
	}

	public static <T> ApiResponse<T> error(String message) {
		return new ApiResponse<>(false, message, null);
	}

	public static <T> ApiResponse<T> error(String message, T data) {
		return new ApiResponse<>(false, message, data);
	}
}

File: fornituras-api-dotnet/src/Fornituras.Api/Common/ApiResponse.cs
namespace Fornituras.Api.Common;

public sealed record ApiResponse<T>(bool Success, string Message, T? Data)
{
    public static ApiResponse<T> Ok(T data, string message = "operation successful") =>
        new(true, message, data);

    public static ApiResponse<T> Error(string message, T? data = default) =>
        new(false, message, data);
}
